### ISSUE TITLE
make the "pause at height" post process  context based

### ISSUE THREAD

**Issue by SchlSE** [Ultimaker/Cura#7198]
**Is your feature request related to a problem? Please describe.**
The Post process list is in my opinion a mess. instead of having one entry for "pause of layer height"
we have four entries that aren't even listed directly after another. And the difference is basically the used commands. If you don't read every entry it is possible to use the false one, resulting in abortet prints ,ect.

**Describe the solution you'd like**
make it context sensitive to the printer and its code setting.

**Describe alternatives you've considered**
make a dropdown  menufor the version in the setting of the process

**Affected users and/or printers**
Basically everyone that uses Boards that doesn't use the marlin code set

**Comment by fieldOfView** [comment]
I agree. Not only is it confusing for end-users which one to pick, but it is also a bit of a nightmare for maintenance, if eg an off-by-one error gets fixed in one flavor but not in the others.

**Comment by fieldOfView** [comment]
I'm working on it here: #7201

**Comment by Ghostkeeper** [comment]
I think most of the development team agrees wholeheartedly with this. It's been a pain to maintain just as well as it's been a pain to use. I'm glad fieldOfView took this on. We also have a ticket on our backlog about this.
